### ISSUE TITLE
More specific moderator privileges

### ISSUE THREAD

**Issue by jacob-g** [futuresight/futurebb#166]
Specifically...
* Viewing IP addresses
* Banning users
* Deleting posts
* Editing posts

Also, change the default admin link for moderators to be towards reports instead of bans.

**Comment by Krett12** [comment]
This isn't what I meant. I meant that there is a menu where you can decide what it means to be a moderator (besides the pre-defined group :P ) . Like a checkbox that says whether or not they can ban users, and another checkbox for whether or not they can edit others' posts, etc.

**Comment by jacob-g** [comment]
That's what I meant by this. The user group page would look something like this:
[ ] moderator privileges
 -> [ ] moderators can view IP addresses
 -> [ ] moderators can ban users
 -> [ ] moderators can delete posts
 -> [ ] moderators can edit posts

**Comment by Krett12** [comment]
That's still not what I meant. I meant that there should be some sort of admin/mod-privileges.php  rather than having it all be a sub-checkboxes of the first one. Otherwise, there wouldn't be much point in the first one!

**Comment by jacob-g** [comment]
![user groups - futurebb 1 4 official prototype 2016-01-10 10-49-19](https://cloud.githubusercontent.com/assets/1960707/12222347/d773e896-b787-11e5-97ee-d77e728c8ef6.png)

The first checkbox still grants users access to a few basic things, but you can be more specific about which permissions they do have by using the subsequent checkboxes. Having it in a separate file is overcomplicating things, and doing it this way allows you to have multiple moderator groups with different permissions (e.g. giving users you trust more the ability to ban users, while still giving mods you aren't as sure of the ability to do stuff like delete posts).

**Comment by Krett12** [comment]
But if you're going to have seperate checkboxes, then why bother having the first one?

**Comment by jacob-g** [comment]
I just said in the above message that it still grants access to a few basic things.

**Comment by Krett12** [comment]
But the point of having the checkboxes is that not everyone agrees on what are "basic things" e.g. banning people. In FluxBB, mods being able to ban people is a seperate option. On the music & running forum, when I was taking a vacation, I made derpmeup a moderator. Unfortunately, I had to revoke it once I found out that he could see IPs. Everyone who signed up on that since signed up with the expectation that only I would be able to see their IP. What are the "basic things" exactly?. According to the picture, they don't count being able to edit & delete everybody's posts. What are they then? It is confusing.
